sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosPatch.java

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.patch.PatchOperation;
+import com.azure.cosmos.implementation.patch.PatchOperationCore;
+import com.azure.cosmos.implementation.patch.PatchOperationType;
+import com.azure.cosmos.util.Beta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Grammar based on RFC: https://tools.ietf.org/html/rfc6902#section-4.1
+ *
+ * Contains a list of Patch operations to be applied on an item. It is applied in an atomic manner and the operations
+ * grammar follows above RFC.
+ *
+ * This can be executed in 3 ways:
+ *  1. Passing this to container in container.patchItem() which requires the id of the item to be patched, partition
+ *      key, the CosmosPatch instance, any CosmosItemRequestOptions and the class type for which response will be parsed.
+ *  2. Add CosmosPatch instance in TransactionalBatch using batch.patchItemOperation() which requires the id of the item
+ *      to be patched, cosmos patch instance and TransactionalBatchItemRequestOptions(if-any) and follow remaining
```

Comment:
Nice - i appreciate that you added patch support for bulk and batch already - this will allow nice new scenarios in the Spark connector for example!